### PR TITLE
Vn madgwick

### LINKED ISSUE #41: Madgwick calibration for VN-300
Currently the calibration matrices (offset, nonlinearity, etc) for the madgwick filter are at default values. We will probably need to change these to match the specs of the VN's IMU, in order to get the most accurate result. We also need to test that the madgwick filter does actually work with VN data at all.

### PR DESCRIPTION
**Pull request by Joe-Joe-Joe-Joe**
Closes #41 

I have tested with the state estimation thread slowed down to match the current data rate, which as I understand it is 2 Hz, 1 Hz on state estimation seemed to play nice. I've added print statements that work with Teleplot, which you should get. here is a nice set of plots, with me just leaving the VN sitting on the table

image

I've added proper scaling of the VN output into the units Fusion expects at the parsing step, and the raw values from the IMU are so good already (aside from gyro drift which the filter does internal compensation for), so I don't think the accel/gyro configuration matrices really need to be modified. Or at least, its not worth the time to conduct an analysis on whether that makes a meaningful difference. Magnetometer is a separate thing with its own empirical compensation parameters, see #57 

Next steps for this would be to reconfigure the VN to output raw data at a higher rate, and see if we can boost state estimation up to match.

Finally, I played around a bit with the `#defines` so that we hopefully don't run into any issues where we load a version that uses the ICM instead of the VN on the fight board.

### CODE REVIEW

**Comment by Joe-Joe-Joe-Joe** [comment]
probably need to restest this again after the merge :face:

**Comment by Joe-Joe-Joe-Joe** [comment]
still works :)

File: STM32Cube/Tasks/state_estimation.c [hunk 1]
```diff
@@ -15,35 +15,39 @@
 #include "printf.h"
 #include "state_estimation.h"
 
-#define TASK_DELAY_TICKS 20 // TODO: what is actual delay time?
-#define SAMPLE_RATE 100 // replace this with actual sample rate
-#define USE_ICM 0
+#define SAMPLE_RATE_HZ 1 //Hz; Need to set to match VN data rate
```
Comment (celery6) [inline STM32Cube/Tasks/state_estimation.c:18]: 1 Hz ??

File: STM32Cube/Tasks/state_estimation.c [hunk 2]
```diff
@@ -151,43 +156,48 @@ void stateEstTask(void *arguments) {
...
-        //Push acceleration values for debugging
-        const FusionVector earth = FusionAhrsGetEarthAcceleration(&ahrs);
-        logDebug("stateEst", "AccelX %d, AccelY %d, AccelZ %d", earth.axis.x, earth.axis.y, earth.axis.z);
+			logInfo("stateEst", "EuRoll %f, EuPitch %f, EuYaw %f", euler.angle.roll, euler.angle.pitch, euler.angle.yaw);
+			//printf_(">EuRoll:%.2f\n>EuPitch:%.2f\n>EuYaw:%.2f\n", euler.angle.roll, euler.angle.pitch, euler.angle.yaw);
 
-        vTaskDelayUntil(&xLastWakeTime, TASK_DELAY_TICKS);
+			//Push acceleration values for debugging
+			const FusionVector earth = FusionAhrsGetEarthAcceleration(&ahrs);
+			logDebug("stateEst", "AccelX %d, AccelY %d, AccelZ %d", earth.axis.x, earth.axis.y, earth.axis.z);
```
[the remark this answers is not in the document]
Comment (Joe-Joe-Joe-Joe) [reply STM32Cube/Tasks/state_estimation.c:197]: it should not
honestly it should be removed entirely

File: STM32Cube/Tasks/state_estimation.c (hunk 1 above)
Comment (Joe-Joe-Joe-Joe) [reply STM32Cube/Tasks/state_estimation.c:18]: Yes. Until we do some more testing with bumping the VN output rate, this is the config that works (VN is outputting at nominal 2 Hz but I ran into issues with readings getting dropped so this is what works consistently)

**Review by celery6 — APPROVED** [review]

**Pull request merged**